detect_secrets/plugins/keyword.py

```diff
@@ -49,30 +49,31 @@
 BLACKLIST_REGEX = re.compile(
     # Followed by double-quotes and a semi-colon
     # e.g. private_key "something";
+    # e.g. private_key 'something';
     r'|'.join(
-        '(.*?){}(.*?)"(.*?)";'.format(
+        '{}(.*?)("|\')(\\S*?)(\'|");'.format(
```

Comment:
We've got a bit too many capturing groups here, which will make `secret_generator` produce unexpected results.

You can use `(?:<>)` to indicate a non-capturing group.